frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+  });
+}
+
+function processLocalesStatistic(serverResponse) {
+  _state['locales'] = {};
+  _state['localeOptions'] = [];
+  _state['srcLocale'] = serverResponse['srcLocale'];
+  resetCache();
+
+  _.forEach(serverResponse['transLocale'], function(transLocale) {
+    _state['locales'][transLocale.locale.localeId] = transLocale;
+    _state['localeOptions'].push({
+      value: transLocale.locale.localeId,
+      label: `${transLocale.locale.displayName} - (${transLocale.numberOfTerms})`
+    });
+  });
+  return _state;
+}
+
+function glossaryAPIUrl(srcLocaleId, transLocale) {
+  var page = _state['page'], filter = _state['filter'];
+
+  var url = Configs.baseUrl + "/glossary/src/" + srcLocaleId;
+
+  if(!StringUtils.isEmptyOrNull(transLocale)) {
+    url = url + "/trans/" + transLocale;
+  }
+  url = url + Configs.urlPostfix + "?page=" + page + "&sizePerPage=" + SIZE_PER_PAGE;
+
+  if(!StringUtils.isEmptyOrNull(filter)) {
+    url = url + "&filter=" + filter;
+  }
+  return url + generateSortOrderParam();
+}
+
+
+function generateSortOrderParam() {
+  var params = [];
+  _.forOwn(_state['sort'], function (value, field) {
+    var param = value === true ? field : "-" + field;
+    params.push(param);
+  });
+
+  return _.size(params) > 0 ? "&sort=" + params.join() : '';
+}
+
+function loadGlossaryByLocale () {
+  var srcLocale = _state['srcLocale'],
+    selectedTransLocaleId = _state['selectedTransLocale'];
+
+  if(!_.isUndefined(srcLocale) && !_.isNull(srcLocale)) {
+    var url = glossaryAPIUrl(srcLocale.locale.localeId, selectedTransLocaleId);
+
+    return new Promise(function(resolve, reject) {
+      Request.get(url)
+        .set("Cache-Control", "no-cache, no-store, must-revalidate")
+        .set('Accept', 'application/json')
+        .set("Pragma", "no-cache")
+        .set("Expires", 0)
+        .end(function (err, res) {
+          if(err != null && err.error === true) {
+            console.error(url, err);
+          }
+          if(res != null) {
+            if (res.error) {
+              console.error(url, res.status, res.error.toString());
+              reject(Error(res.error.toString()));
+            } else {
+              resolve(res['body']);
+            }
+          }
+          resolve(null);
+        });
+    });
+  }
+}
+
+function canAddNewEntry () {
+ return Configs.data.permission.insertGlossary;
+}
+
+function canUpdateEntry() {
+  return Configs.data.permission.updateGlossary;
+}
+
+function processGlossaryList(serverResponse) {
+  if(serverResponse === null) {
+    return _state;
+  }
+  var transLocaleId = _state['selectedTransLocale'],
+    page = _state['page'];
+
+  _state['totalCount'] = serverResponse.totalCount;
+
+  var startIndex = ((_state['page'] -1) * SIZE_PER_PAGE);
+
+  for (var i = startIndex; i < _state['totalCount']; i++) {
+    _state['glossaryResId'].push(null);
+  }
```

Comment:
What's going on here? Filling an array with a bunch of `null`s? This definitely needs some kind of explanation.